explainer.md

```diff
@@ -161,119 +160,134 @@ user agent.
 
-1. An entire origin (e.g., when you just want to set an unread count for the
-   entire site).
-2. All pages whose URL [starts
-   with](https://www.w3.org/TR/appmanifest/#dfn-within-scope) a given path
-   (e.g., if a site is organised into "projects", each with their own status
-   indicator).
-3. All pages with a specific URL path (e.g., if each page has a separate status
-   indicator, but all tabs open at a certain URL should share status).
-4. A specific document (e.g., if many tabs can be open with URLs on the same
-   path, but might have different status, due to either a different query
-   string, or different dynamic context inside the document).
+The reason we have separate APIs is that they have very different concerns which
+are now properly separated:
 
-Number 3 and 4 seem very similar, but architecturally are very different. In
-3, the badge is associated with a URL (and any current or future documents open
-at that URL will show that badge), whereas in 4, the badge is associated with a
-particular document — other tabs open to the same URL will not share the badge.
+* The URL scoping and push message complexity only applies to the scope API.
+* The presence of the document API itself indicates whether the tab will be
+  badged in the UI (obviating the need for a dedicated feature detection method
+  specifically for tab badging).
 
-What this boils down to is two different modes:
+Other reasons to separate:
 
-* Badging a particular URL scope, applied to the set of URLs that start with the
-  same path. This covers 1, 2 and 3 above.
-* Badging a particular document, applied to only the browser tab containing the
-  document.
+* Sites that only want to badge their app icon or only want to badge their tab
+  can just concern themselves with the relevant API.
+* The document API can potentially support badges that aren't supported by host
+  operating systems, such as arbitrary Unicode characters.
 
-The API for requesting to badge the current document, as opposed to a URL set,
-is still under consideration. It may simply be a separate option member, e.g.:
+User agents may provide either or both of the two APIs, so sites should
+feature-detect them individually. In particular, this allows you to fall back to
+showing your own badge in the page favicon or title if the
+`Badge.setForDocument` API is not available (regardless of the availability of
+the scope API). This is a complete example for a site that wants to set the
+badge on both the current document and for apps/handles across the origin:
 
 ```js
-Badge.set(getUnreadCount(), {scopeDocument: true});
+// Should be called whenever the unread count changes (new mail arrives, or mail
+// is marked read/unread).
+function unreadCountChanged(newUnreadCount) {
+  // Set the "handle" badge, for app icons and links. This has a global and
+  // semi-permanent effect, outliving the current document.
+  if (Badge && Badge.setForScope) {
+    Badge.setForScope(newUnreadCount);
```

Comment:
Probably separate sections. We'd try to have them listed separately on caniuse but I don't want to write two different specs because they're closely related.

Reply:
SGTM :)